### PR TITLE
Improved text editor

### LINKED ISSUE #34: Update text editor
ckeditor is quite archaic

### PR DESCRIPTION
**Pull request by krzys-h**
CKEditor was replaced by TinyMCE. A few useful things came out as a result:

* It is now possible to upload images in the workshop page and article editors. Previously, you would have to host them on some external service and provide an url which was not ideal... I also made a script that can download all images referenced by old pages and host them on our server.
* The styles in the editor actually match the ones on the resulting page
* The editor seems more usable in general.

Old editor:
image

New editor:
image

Actual page content:
image

Closes #34

### CODE REVIEW

**Comment by zielmicha** [comment]
Have you ensured that html (with javascript that will steal your cookies) files can't be uploaded? (We *need* to have CSP policy on MEDIA_ROOT anyway.)

**Comment by krzys-h** [comment]
Yes - I used ImageField so that Django runs its usual checks to validate that the file is actually an image and verified that it works. Additionally, only trusted users like article editors and lecturers with accepted workshops have access to the upload functionality.

File: wwwapp/management/commands/download_remote_images.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import wwwapp.settings as settings
+from wwwapp.models import Workshop, Article
+
+
+class Command(BaseCommand):
+    help = 'Download all images referenced in <img> tags in old descriptions and upgrade to storing them on our server'
+
+    def add_arguments(self, parser):
+        group = parser.add_mutually_exclusive_group(required=True)
```
Comment (cytadela8) [inline wwwapp/management/commands/download_remote_images.py]: It should be possible to specify in parser that one of `force` and `dryrun` is required

**Comment by krzys-h** [comment]
> I would be more worried about javascript beeing included in the html page itself. CSP policies will help, but proper sanitizations is the best way to go.

Bleach already handles that, you can't use <script> tags or anything like that

File: wwwapp/management/commands/download_remote_images.py (hunk 1 above)
Comment (krzys-h) [reply wwwapp/management/commands/download_remote_images.py]: Good point

**Comment by cytadela8** [comment]
@krzys-h You are right. + we should fix #53

**Comment by cytadela8** [comment]
Add `Pillow` to requirements.txt

**Comment by krzys-h** [comment]
That's interesting, I don't remember installing Pillow manually but I'll check

**Comment by krzys-h** [comment]
Aaand you were right! I have no idea how Pillow ended up being already installed in my venv, even my shell history doesn't show the install command anywhere...

File: wwwapp/management/commands/download_remote_images.py (hunk 1 above)
Comment (krzys-h) [reply wwwapp/management/commands/download_remote_images.py]: Actually... that is the whole point of this being a `mutually_exclusive_group` XD That's why the condition in handle() is an assert and not an exception or error message

**Pull request merged**